src/plugins/sailfishoswizards/desktopeditorwidget.cpp

```diff
@@ -0,0 +1,358 @@
+        m_fileContent.remove("Icon=%{Icon}\n");
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/*!
+ * \brief Clears all entered data from the table model.
+ */
+void DesktopEditorWidget::clearTranslations()
+{
+    m_tableModel->clearTranslations();
+}
+
+/*!
+ * \brief Returns list of the entered icon paths from the icon page.
+ */
+QHash<QString, QString> DesktopEditorWidget::iconPaths() const
+{
+    QHash<QString, QString> paths;
+    paths.insert("General", m_iconPage->Ui().iconPath->text());
+    paths.insert("86x86", m_iconPage->Ui().icon86Path->text());
+    paths.insert("108x108", m_iconPage->Ui().icon108Path->text());
+    paths.insert("128x128", m_iconPage->Ui().icon128Path->text());
+    paths.insert("172x172", m_iconPage->Ui().icon172Path->text());
+    return paths;
+}
+
+/*!
+ * \brief Reloads icon page. Writes all specific icon paths.
+ */
+void DesktopEditorWidget::reloadIconPage(QString applicationPath, int iconsIndex)
+{
+    m_iconPage->Ui().iconPath->clear();
+    m_iconPage->Ui().manualInputCheck->setCheckState(Qt::CheckState::Checked);
+    m_iconPage->Ui().icon86Path->setText(applicationPath + "/86x86/" + QDir(
+                                             applicationPath).dirName() +
+                                         "_" + QString::number(iconsIndex) + ".png");
+    m_iconPage->Ui().icon108Path->setText(applicationPath + "/108x108/" + QDir(
+                                              applicationPath).dirName() +
+                                          "_" + QString::number(iconsIndex) + ".png");
+    m_iconPage->Ui().icon128Path->setText(applicationPath + "/128x128/" + QDir(
+                                              applicationPath).dirName() +
+                                          "_" + QString::number(iconsIndex) + ".png");
+    m_iconPage->Ui().icon172Path->setText(applicationPath + "/172x172/" + QDir(
+                                              applicationPath).dirName() +
+                                          "_" + QString::number(iconsIndex) + ".png");
+}
+
+/*!
+ * \brief Checks if the entered icons exist, have correct extension and resolution.
+ * Shows error message if any of this conditions is not satisfied.
+ * Returns \c true if all the validations passed, otherwise returns \c false.
+ */
+bool DesktopEditorWidget::validateIcons()
+{
+    QMimeDatabase mimeDB;
+    QString errorString = "";
+    if (m_iconPage->Ui().manualInputCheck->isChecked()) {
+        QList<QLineEdit *> pathEdits = {m_iconPage->Ui().icon86Path,
+                                        m_iconPage->Ui().icon108Path,
+                                        m_iconPage->Ui().icon128Path,
+                                        m_iconPage->Ui().icon172Path
+                                       };
+        if (QPixmap(m_iconPage->Ui().icon86Path->text()).height() != 86
+                || QPixmap(m_iconPage->Ui().icon86Path->text()).height() != 86)
```

Comment:
I'm quite sure one of the lines above should say "width"

Reply:
Fixed